internal/ops/wstrust/wstrust_test.go

```diff
@@ -0,0 +1,308 @@
+			giveResp: wstrust.SAMLDefinitions{
+				Body: wstrust.Body{
+					RequestSecurityTokenResponseCollection: wstrust.RequestSecurityTokenResponseCollection{
+						RequestSecurityTokenResponse: []wstrust.RequestSecurityTokenResponse{
+							{
+								RequestedSecurityToken: wstrust.RequestedSecurityToken{
+									Assertion: wstrust.Assertion{
+										Text: "hello",
+										XMLName: xml.Name{
+											Local: "Assertion",
+										},
+										Saml: samlv1Assertion,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			desc:              "Success: SAMLV1 assertion with AuthorizationTypeWindowsIntegratedAuth",
+			endpoint:          wstrust.Endpoint{EndpointVersion: wstrust.Trust13, URL: "upEndpoint"},
+			action:            SoapActionDefault,
+			authorizationType: msalbase.AuthorizationTypeWindowsIntegratedAuth,
+			body:              "<s:Envelope xmlns:s=\"http://www.w3.org/2003/05/soap-envelope\" xmlns:wsa=\"http://www.w3.org/2005/08/addressing\" xmlns:wsu=\"http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd\"><s:Header><wsa:Action s:mustUnderstand=\"1\">http://docs.oasis-open.org/ws-sx/ws-trust/200512/RST/Issue</wsa:Action><wsa:messageID>urn:uuid:fb8ec65b-f117-468f-b4e8-50c5e802affe</wsa:messageID><wsa:ReplyTo><wsa:Address>http://www.w3.org/2005/08/addressing/anonymous</wsa:Address></wsa:ReplyTo><wsa:To s:mustUnderstand=\"1\">upEndpoint</wsa:To><wsse:Security s:mustUnderstand=\"\" xmlns:wsse=\"\"><wsu:Timestamp wsu:Id=\"\"><wsu:Created></wsu:Created><wsu:Expires></wsu:Expires></wsu:Timestamp><wsse:UsernameToken wsu:Id=\"\"><wsse:Username></wsse:Username><wsse:Password></wsse:Password></wsse:UsernameToken></wsse:Security></s:Header><s:Body><wst:RequestSecurityToken xmlns:wst=\"http://docs.oasis-open.org/ws-sx/ws-trust/200512\"><wsp:AppliesTo xmlns:wsp=\"http://schemas.xmlsoap.org/ws/2004/09/policy\"><wsa:EndpointReference><wsa:Address>urn</wsa:Address></wsa:EndpointReference></wsp:AppliesTo><wst:KeyType>http://docs.oasis-open.org/ws-sx/ws-trust/200512/Bearer</wst:KeyType><wst:RequestType>http://docs.oasis-open.org/ws-sx/ws-trust/200512/Issue</wst:RequestType></wst:RequestSecurityToken></s:Body></s:Envelope>",
+			giveResp: wstrust.SAMLDefinitions{
+				Body: wstrust.Body{
+					RequestSecurityTokenResponseCollection: wstrust.RequestSecurityTokenResponseCollection{
+						RequestSecurityTokenResponse: []wstrust.RequestSecurityTokenResponse{
+							{
+								RequestedSecurityToken: wstrust.RequestedSecurityToken{
+									Assertion: wstrust.Assertion{
+										Text: "hello",
+										XMLName: xml.Name{
+											Local: "Assertion",
+										},
+										Saml: samlv1Assertion,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		fake := &fakeXMLCaller{err: test.commErr, giveResp: test.giveResp}
+		client := Client{Comm: fake}
+
+		authParams.AuthorizationType = test.authorizationType
```

Comment:
Can we get a test for AuthorizationType == AuthorizationTypeUsernamePassword here as well?